scalagen/src/test/scala/org/scalameta/scalagen/Examples.scala

```diff
@@ -74,21 +72,11 @@ object Freeish extends TransmutationGenerator("Freeish") {
 }
 
 object DeleteMe extends TransmutationGenerator("DeleteMe") {
-  override def transmute(t: Defn.Trait): List[Defn] = Nil
-  override def transmute(c: Defn.Class): List[Stat] = Nil
-  override def transmute(t: Defn.Type): List[Stat] = Nil
-  override def transmute(o: Defn.Object): List[Stat] = Nil
-  override def transmute(d: Defn.Def): List[Stat] = Nil
-  override def transmute(v: Defn.Val): List[Stat] = Nil
-  override def transmute(v: Defn.Var): List[Stat] = Nil
-  override def transmute(v: Decl.Var): List[Stat] = Nil
-  override def transmute(v: Decl.Val): List[Stat] = Nil
-  override def transmute(d: Decl.Def): List[Stat] = Nil
-  override def transmute(t: Decl.Type): List[Stat] = Nil
+  override def transmute[A <: Tree : StatReplacer](a: A): List[Stat] = Nil
 }
 
 object Abort extends ExtensionGenerator("Abort") {
-  override def extend(c: Defn.Class): List[Stat] =
+  override def extend[A <: Tree : StatExtractor](a: A): List[Stat] =
```

Comment:
Keep in mind that the implicit evidence complicates runtime reflection, in case we ever want to reflectively invoke the extend methods.

Reply:
If I my plan works out. we wont use reflection at all.